runtime/lua/vim/fcnotify.lua

```diff
@@ -0,0 +1,224 @@
+--- Ignores all the buffers that aren't listed, or have a buf id
+--- less than 0.
+--- Ignores the [NO NAME] buffer.
+---
+--@param fname: The validity of this buffer will be checked.
+local function valid_buf(fname)
+  if fname == '' then
+    return false
+  end
+
+  local bufnr = vim.api.nvim_call_function('bufnr', {fname})
+
+  if bufnr < 0 then
+    return false
+  end
+
+  local buflisted = vim.api.nvim_buf_get_option(bufnr, 'buflisted')
+  local buftype = vim.api.nvim_buf_get_option(bufnr, 'buftype')
+
+  if not buflisted or buftype == 'nofile' or buftype == 'quickfix' then
+    return false
+  end
+  return true
+end
+
+local check_handle = uv.new_check()
+check_handle:start(vim.schedule_wrap(check_notifications))
+
+--- Creates and initializes a new watcher object with the given filename.
+---
+--@param fname: (required, string) The path that the watcher should watch.
+function Watcher:new(fname)
+  vim.validate{fname = {fname, 'string', false}}
+  -- get full path name for the file
+  local ffname = vim.api.nvim_call_function('fnamemodify', {fname, ':p'})
+  w = {bufnr = vim.api.nvim_call_function('bufnr', {fname}),
+       fname = fname, ffname = ffname, handle = nil,
+       paused = false, pending_notifs = false}
+  setmetatable(w, self)
+  self.__index = self
+  return w
+end
+
+local function start_poll(fname)
+  WatcherList[fname].handle = uv.new_fs_poll()
+  WatcherList[fname].handle:start(WatcherList[fname].ffname, 200, function(...)
+    WatcherList[fname]:poll_change(...)
+  end)
+end
+
+--- Starts the watcher
+---
+--@param self: (required, table) The watcher object on which start was called.
+local function start_watch(fname)
+  WatcherList[fname].handle = uv.new_fs_event()
+  WatcherList[fname].handle:start(WatcherList[fname].ffname, {}, function(...)
+    WatcherList[fname]:watch_change(...)
+  end)
+end
+
+function Watcher:start()
+  if self.handle and not self.handle:is_closing() then
+    self.handle:close()
+  end
+
+  commence(self.fname)
+end
+
+function Watcher.init_poll()
+  commence = start_poll
+end
+
+function Watcher.init_watcher()
+  commence = start_watch
+end
+
+--- Stops the watcher and closes the handle.
+---
+--@param self: (required, table) The watcher object on which stop was called.
+function Watcher:stop()
+  self.handle:stop()
+
+  -- close the handle altogether, for windows.
+  if self.handle:is_closing() then
+    return
+  end
+  self.handle:close()
+end
+
+--- Callback for watcher handle. Marks a watcher as having pending
+--- notifications. The nature of notification is determined while
+--- responding to the notification.
+---
+--@param err: (string) Error if any occured during the execution of the callback.
+---
+--@param fname: (string) The file for which the notification was received. (Not
+---             very reliable.)
+---
+--@param event: (table) The type of event recieved for a file.
+function Watcher:watch_change(err, fname, event)
+  -- To satisfy the lua linter.
```

Comment:
You can just put the param name as `_` to silence warnings about unused args if you don't want them.

Reply:
That helped. Thanks!